Tests/Mocks/BillingFormTextFieldCellMockDelegate.swift

```diff
@@ -0,0 +1,37 @@
+import UIKit
+@testable import Frames
+
+class BillingFormTextFieldCellMockDelegate: BillingFormTextFieldCellDelegate {
+    
+    var textFieldShouldBeginEditingCalledTimes = 0
+    var textFieldShouldBeginEditingLastCalledWithTextField: UITextField?
```

Comment:
As all software unit tests can grow heavily quickly, so I would propose to join the counter and the tracker property into an array of called properties.

Where you currently record things like:
```
calledTimes += 1
lastCalled = textField
```

you could instead record as:
```
calledTextFields.append(textField)
```

This now allows you to test counter as well as verify every single passed object if you did expect more than one to be passed. And you only need 1 property, not 2.

Reply:
I think, i have a better idea. we can start using Sourcery. what do you think?
https://github.com/krzysztofzablocki/Sourcery

Reply:
I have only seen Sorcery used in 2 projects so far and they were actually teams that were not very well focussed on native, so they tried to use approaches that could be easily used in an analogue project on a different project. Personally I am a fan of learning eco stem well first and using available tools correctly, before trading off native to more cross team.

Having said that, I don't want to dismiss something only based on my personal historic bad experience. I think it could be very valuable to understand how you see it and for us as a team to discuss on it more.

For now, this is unit test, not live code, we could keep as is, but would also like Deepesh or Harry to confirm they see no problem with it. For me, I am always weary when merging some pattern that can immediately be improved on that it establishes the wrong models for others to potentially copy in the future. But at the same time, testing is better than no testing, so 🤷‍♂️

Reply:
I think, this requires a tech meeting for this update to choose best approach for future work.

Reply:
I have also [posted on Slack](https://checkout.slack.com/archives/C038WRNDKTM/p1651661700347499) a link to an article describing exactly the flow you're going through. The creator starts with your exact approach, then improves it to what I suggested and I think explains quite well the transition.

Would really like your input on that, regardless of the changes (/ lack of) in this particular case 😉